src/test/ui/const-generics/min_const_generics/forbid-non-static-lifetimes.rs

```diff
@@ -0,0 +1,35 @@
+// revisions: full min
+// rustc-env:RUST_BACKTRACE=0
+// normalize-stderr-test "note: rustc 1.* running on .*" -> "note: rustc VERSION running on TARGET"
+// normalize-stderr-test "note: compiler flags: .*" -> "note: compiler flags: FLAGS"
+// normalize-stderr-test "/universal_regions.rs:[0-9]+:[0-9]+" -> "/universal_regions.rs:LL:CC"
+//[full]failure-status: 101
+#![cfg_attr(full, feature(const_generics))]
+#![cfg_attr(full, allow(incomplete_features))]
+#![cfg_attr(min, feature(min_const_generics))]
+
+// This test checks that non-static lifetimes are prohibited under `min_const_generics`. It
+// currently emits an error with `min_const_generics` and an ICE with `const_generics`.
```

Comment:
I think we should just add the `min_const_generics` test, but leave the issue that's not fixed on `const_generics` open with an appropriate label.